### PR TITLE
Throw aggregate error for `npmName.many()`

### PR DESCRIPTION
**Pull request by baile320**
Fixes https://github.com/sindresorhus/npm-name/issues/29

### CODE REVIEW

**Comment by baile320** [comment]
Hmm... there's an error:

  index.test-d.ts:14:33
  ✖  14:33  Top-level await expressions are only allowed when the module option is set to esnext or system, and the target option is set to es2017 or higher.

I get this when running locally on the master branch as well. I'm not what the preferred fix is. Wrapping the code in index.test-d.ts in an IIFE resolves the issue. Any preferences?

File: index.js [hunk 1]
```diff
@@ -90,8 +91,12 @@ module.exports.many = async (names, options = {}) => {
 		throw new Error('The `registryUrl` option must be a valid string URL');
 	}
 
-	const result = await Promise.all(names.map(name => request(name, options)));
-	return new Map(zip(names, result));
+	try {
```
Comment (sindresorhus) [inline index.js]: This try/catch is moot as you're just catching and throwing again.

File: test.js [hunk 2]
```diff
@@ -67,3 +67,19 @@ test('throws when package name is invalid', async t => {
 - name cannot start with an underscore`
 	});
 });
+
+test('should return an interable error capturing multiple errors when appropriate', async t => {
```
Comment (sindresorhus) [inline test.js]:
```suggestion
test('should return an iterable error capturing multiple errors when appropriate', async t => {
```

File: test.js [hunk 3]
```diff
@@ -67,3 +67,19 @@ test('throws when package name is invalid', async t => {
...
+	const name1 = 'chalk'; // False
+	const name2 = uniqueString(); // True
+	const name3 = '_ABC'; // Error
+	const name4 = 'CapitalsAreBad'; // Error
+
+	try {
```
Comment (sindresorhus) [inline test.js]: This should use a `t.throwsAsync` assertion.

File: test.js [hunk 4]
```diff
@@ -67,3 +67,19 @@ test('throws when package name is invalid', async t => {
...
+		await npmName.many([name1, name2, name3, name4]);
+	} catch (error) {
+		const errors = [...error];
+		t.true(errors.length === 2);
```
Comment (sindresorhus) [inline test.js]:
```suggestion
		t.is(errors.length, 2);
```

**Comment by sindresorhus** [comment]
> I get this when running locally on the master branch as well. I'm not what the preferred fix is. Wrapping the code in index.test-d.ts in an IIFE resolves the issue. Any preferences?

Just ignore that. It will be fixed when I upgrade dependencies after this is merged.

**Comment by baile320** [comment]
I made those changes. Let me know if you have any other comments... thanks for your patience!

**Comment by sindresorhus** [comment]
Looks good. Thanks :)

**Pull request merged**